runtime/environment.go

```diff
@@ -592,111 +594,52 @@ func (e *interpreterEnvironment) newOnRecordTraceHandler() interpreter.OnRecordT
 	}
 }
 
-func (e *interpreterEnvironment) newHashHandler() interpreter.HashHandlerFunc {
-	return func(
-		inter *interpreter.Interpreter,
-		locationRange interpreter.LocationRange,
-		dataValue *interpreter.ArrayValue,
-		tagValue *interpreter.StringValue,
-		hashAlgorithmValue interpreter.MemberAccessibleValue,
-	) *interpreter.ArrayValue {
-		data, err := interpreter.ByteArrayValueToByteSlice(inter, dataValue)
-		if err != nil {
-			panic(errors.NewUnexpectedError("failed to get data. %w", err))
-		}
-
-		var tag string
-		if tagValue != nil {
-			tag = tagValue.Str
-		}
-
-		hashAlgorithm := stdlib.NewHashAlgorithmFromValue(inter, locationRange, hashAlgorithmValue)
-
-		var result []byte
-		wrapPanic(func() {
-			result, err = e.runtimeInterface.Hash(data, tag, hashAlgorithm)
-		})
-		if err != nil {
-			panic(err)
-		}
-		return interpreter.ByteSliceToByteArrayValue(inter, result)
-	}
-}
-
 func (e *interpreterEnvironment) newPublicAccountHandler() interpreter.PublicAccountHandlerFunc {
 	return func(address interpreter.AddressValue) interpreter.Value {
 		return stdlib.NewPublicAccountValue(e, e, address)
 	}
 }
 
-func (e *interpreterEnvironment) newSignatureVerificationHandler() interpreter.SignatureVerificationHandlerFunc {
-	return func(
-		inter *interpreter.Interpreter,
-		locationRange interpreter.LocationRange,
-		signatureValue *interpreter.ArrayValue,
-		signedDataValue *interpreter.ArrayValue,
-		domainSeparationTagValue *interpreter.StringValue,
-		hashAlgorithmValue *interpreter.SimpleCompositeValue,
-		publicKeyValue interpreter.MemberAccessibleValue,
-	) interpreter.BoolValue {
-
-		signature, err := interpreter.ByteArrayValueToByteSlice(inter, signatureValue)
-		if err != nil {
-			panic(errors.NewUnexpectedError("failed to get signature. %w", err))
-		}
-
-		signedData, err := interpreter.ByteArrayValueToByteSlice(inter, signedDataValue)
-		if err != nil {
-			panic(errors.NewUnexpectedError("failed to get signed data. %w", err))
-		}
-
-		domainSeparationTag := domainSeparationTagValue.Str
-
-		hashAlgorithm := stdlib.NewHashAlgorithmFromValue(inter, locationRange, hashAlgorithmValue)
-
-		publicKey, err := stdlib.NewPublicKeyFromValue(inter, locationRange, publicKeyValue)
-		if err != nil {
-			return false
-		}
-
-		var valid bool
-		wrapPanic(func() {
-			valid, err = e.runtimeInterface.VerifySignature(
-				signature,
-				domainSeparationTag,
-				signedData,
-				publicKey.PublicKey,
-				publicKey.SignAlgo,
-				hashAlgorithm,
-			)
-		})
+func (e *interpreterEnvironment) ValidatePublicKey(publicKey *stdlib.PublicKey) error {
```

Comment:
New functions just delegate to the runtime interface, like the other stdlib functions of the environment